tests/drivers/random/randomnumgen_basic_api/src/main.c

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2017 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <random.h>
+#include <ztest.h>
+
+/*
+ * @addtogroup t_randomnumgen_basic_api
+ * @{
+ * @defgroup t_random_get_entropy test_random_get_entropy
+ * @brief TestPurpose: verify Random number generator works
+ * @details
+ * - Test Steps
+ *   -# Read random numbers from RNG driver.
+ *   -# Verify whether buffer overflow occured or not.
+ *   -# Verify whether buffer completly filled or not.
+ * - Expected Results
+ *   -# Random number should be generated.
+ * @}
+ */
+
+#define BUFFER_LENGTH 10
+#define RECHECK_RANDOM_ENTROPY 0x10
+
+static int random_entropy(struct device *dev, char *buffer, char num)
+{
+	int ret, i;
+	int count = 0;
+
+	memset((char *)buffer, (char)num, BUFFER_LENGTH);
```

Comment:
No need to cast these to `char *` and `char`: any pointer can be cast to a `void *` (first argument type of memset), and chars are promoted to integers (second argument to memset).  In any case, these casts are not helping anything: the types of `buffer` and `num` expressions are already `char *` and `char` respectively